libdrgn/program.c

```diff
@@ -264,6 +264,7 @@ drgn_program_set_core_dump_fd_internal(struct drgn_program *prog, int fd,
 	const char *vmcoreinfo_note = NULL;
 	size_t vmcoreinfo_size = 0;
 	bool have_nt_taskstruct = false, is_proc_kcore;
+	bool have_vmcoreinfo = prog->vmcoreinfo.raw;
```

Comment:
I just noticed that `drgn_program_set_core_dump_fd_internal()` and `drgn_program_set_kdump()` both free `prog->vmcoreinfo.raw` in the error case, so the error handling needs to be updated to handle the case that the vmcoreinfo was already set.

Reply:
Looks like we're getting `have_vmcoreinfo` and `had_vmcoreinfo` 🙃